src/include/recovery/log_record.h

```diff
@@ -54,9 +54,9 @@ enum class LogRecordType {
  * | HEADER | tuple_rid | tuple_size | old_tuple_data | tuple_size | new_tuple_data |
  *-----------------------------------------------------------------------------------
  * For new page type log record
- *--------------------------
- * | HEADER | prev_page_id |
- *--------------------------
+ *------------------------------------
+ * | HEADER | prev_page_id | page_id |
```

Comment:
Actually, I think that the original comment was good.

Considering that the serialization format is used for recovery only, then when you're trying to recover from the last checkpoint, you don't have to worry about the new page id, as the disk manager will do that for you.

There is no point that you request to allocate a new page with a specific page id

Reply:
First, this is a description of the log record structure, rather than which part of the structure should be used in your program. It has to be a complete description. I think the error has been introduced in previous versions of this homework. The previous version of the 'NEWPAGE' log record lack the page_id part, it was added later but the comment remains unchanged and mismatches the actual log record structure. 
Second, we are not supposed to discuss how homework details here. Generally, I think when you redo a NEWPAGE, it was already in the disk. You should fetch it from the disk and clean it.